### PR TITLE
Framework tests with monoscopic backend.

### PR DESCRIPTION
**Pull request by sushantojal**
Monoscopic screenshots with Vk and GL.
Removed monoscopic dependencies from within the framework. (such as boolean monoscopic flags being passed around).
Depends on PR 264 in tests repo.

GearVRF DCO signed off by: Sushant Ojal sushant.o@samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
Needs rebase

**Comment by sushantojal** [comment]
working on this, will also remove the VkRenderTextureOnScreen and VkRenderTextureOffScreen classes, since there is no difference in both.

**Comment by sushantojal** [comment]
Have removed the VkRenderTextureOnScreen and VkRenderTextureOffScreen classes, and also other monoscopic specific hacks we had in the framework, mainly in GVRRenderTarget and GVRRenderTexture.

**Comment by liaxim** [comment]
Needs rebase

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRViewManager.java [hunk 1]
```diff
@@ -688,7 +690,9 @@ protected void captureCenterEye(GVRRenderTarget renderTarget, boolean isMultivie
         // TODO: when we will use multithreading, create new camera using centercamera as we are adding posteffects into it
         final GVRCamera centerCamera = mMainScene.getMainCameraRig().getCenterCamera();
         final GVRMaterial postEffect = new GVRMaterial(this, GVRMaterial.GVRShaderType.VerticalFlip.ID);
-        centerCamera.addPostEffect(postEffect);
+
+        if(!(isVulkanInstance() && mBackend == 6))
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRViewManager.java]: This is a problem. Please look into moving these lines to the monoscopic view manager; you could override captureCenterEye there.

Comment (liaxim) [reply GVRf/Framework/framework/src/main/java/org/gearvrf/GVRViewManager.java]: And lets try to figure out a way that doesn't require isVulkanInstance, is possible. And if it is not possible, let's open an issue to track what I presume is a problem with the design.

**Comment by liaxim** [comment]
Other than the previous note, looks like a nice a cleanup. Thanks.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRViewManager.java (hunk 1 above)
Comment (sushantojal) [reply GVRf/Framework/framework/src/main/java/org/gearvrf/GVRViewManager.java]: Have removed isVulkanInstance and handling the monoscopic cases in the native Vulkan side.

**Comment by liaxim** [comment]
Looks good to me. Planning to merge on Wednesday if there are no comments.

@roshanch ? @apjagdale ?

**Pull request merged**